shell.nix

```diff
@@ -0,0 +1,33 @@
+{ haskell ? true
+, coq     ? true
+, c       ? true
+, nixpkgs ? import <nixpkgs> {}
+}:
+let
+  simplicity      = import ./. {};
+  optional        = nixpkgs.lib.optional;
+  mkDerivation    = nixpkgs.stdenv.mkDerivation;
+
+  haskellDrv =
+    let
+      haskellDevTools = pkgs: with pkgs; [cabal-install hlint hasktags];
+      haskellPkgs     = pkgs: simplicity.haskell.buildInputs ++ haskellDevTools pkgs;
+      haskellDevEnv   = simplicity.haskellPackages.ghcWithPackages haskellPkgs;
+    in
+      mkDerivation {
```

Comment:
for some reason I can't use `haskellDevEnv` directly here, I need to wrap it in another mkDerivation.  `ghcWithPackages` and `nix-shell` interaction continues to confuse me